src/time.rs

```diff
@@ -0,0 +1,79 @@
+//! Types and Functions for temporal manipulation.
+//!
+//! This module provides primitives for setting asynchronous timeouts, intervals, and delays.
+//!
+//! # Organization
+//!
+//! * [`Delay`] and [`Interval`] provide functionality for setting delays and intervals.
+//! * [`FutureExt`] extends Futures with the ability to time-out.
+//! * Other types are return or parameter types for various methods in this module
+//!
+//! [`Delay`]: struct.Delay.html
+//! [`Interval`]: struct.Interval.html
+//! [`FutureExt`]: trait.FutureExt.html
+//!
+//! ## Examples
+//! __Schedule a three-second delay__
+//! ```no_run
+//! # #![feature(async_await)]
+//! # #[runtime::main]
+//! # async fn main() {
+//! use runtime::time::delay_for;
+//! use std::time::{Duration, Instant};
+//!
+//! let start = Instant::now();
+//! let now = delay_for(Duration::from_secs(3)).await;
+//!
+//! let elapsed = now - start;
+//! println!("elapsed: {}s", elapsed.as_secs());
+//! # }
+//! ```
+//!
+//! __Schedule a two-second interval__
+//! ```ignore
+//! # #![feature(async_await)]
+//! # #[runtime::main]
+//! # async fn main() {
+//! # use futures::for_await;
+//! use runtime::time::interval;
+//! use std::time::{Duration, Instant};
+//!
+//! let start = Instant::now();
+//!
+//! #[for_await]
+//! for now in interval(Duration::from_secs(2)) {
+//!     let elapsed = now - start;
+//!     println!("elapsed: {}s", elapsed.as_secs());
+//! }
+//! # }
+//! ```
+
+mod delay;
+mod interval;
+
+pub mod ext;
+
+pub use delay::Delay;
+#[doc(inline)]
+pub use ext::FutureExt;
+pub use interval::Interval;
+
+use std::time::{Duration, Instant};
+
+/// Sleep the current future for the given duration.
```

Comment:
The documentation seems a bit misleading (also below). This gives a `Future` that will delay for the given duration, that will only be `Ready` once that duration has passed. It does not affect any current future (what's the current future?) unless combined with it in one way or another (`await`, `and_then`, ...)

Reply:
Nice and clean API otherwise, I like it :+1: